examples/upgrade-http1x.md

```diff
@@ -0,0 +1,645 @@
+# Upgrading from http@0.2/hyper@0.14 to http@1.x/hyper@1.x
+
+This guide provides a comprehensive walkthrough for upgrading your smithy-rs server applications from http@0.2.x and hyper@0.14 to http@1.x and hyper@1.x.
+
+## Table of Contents
+
+- [Overview](#overview)
+- [Why Upgrade?](#why-upgrade)
+- [Before You Begin](#before-you-begin)
+- [Dependency Updates](#dependency-updates)
+- [Server-Side Changes](#server-side-changes)
+- [Client-Side Changes](#client-side-changes)
+- [Test Infrastructure Updates](#test-infrastructure-updates)
+- [Common Migration Patterns](#common-migration-patterns)
+- [Troubleshooting](#troubleshooting)
+- [Migration Checklist](#migration-checklist)
+
+## Overview
+
+The http and hyper crates have released major version updates (http@1.x and hyper@1.x) with significant API improvements and breaking changes. This guide helps you migrate your smithy-rs server applications to these new versions.
+
+**Key Changes:**
+- http: 0.2.x → 1.x
+- hyper: 0.14.x → 1.x
+- New hyper-util crate for additional utilities
+- aws-smithy-http-server replaces aws-smithy-legacy-http-server
+
+## Why Upgrade?
+
+- **Better Performance**: Hyper 1.x has improved performance and reduced allocations
```

Comment:
is this actually true?

Reply:
Ops! I had completely forgotten to review the file before pushing the commit - sorry.